### PR TITLE
fix `test_orion_full_migration_works_with_data_in_db`

### PR DESCRIPTION
**Pull request by jakekaplan**
### TLDR:
Test was timing out but the error is obscured  because of bad interaction between what this test is doing, it failing and flaky

---
`test_orion_full_migration_works_with_data_in_db` was failing in CI more often then not with an error similar to:
```
sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) table flow_run has no column named work_queue_id
[SQL: INSERT INTO flow_run (id, created, updated, name, state_type, state_name, state_timestamp, run_count, 
expected_start_time, next_scheduled_start_time, start_time, end_time, total_run_time, deployment_id, work_queue_name,
flow_version, parameters, idempotency_key, context, empirical_policy, tags, created_by, infrastructure_pid, auto_scheduled,
flow_id, infrastructure_document_id, parent_task_run_id, state_id, work_queue_id) VALUES (:id, :created, :updated, :name,
:state_type, :state_name, :state_timestamp, :run_count, :expected_start_time, :next_scheduled_start_time, :start_time,
:end_time, :total_run_time, :deployment_id, :work_queue_name, :flow_version, :parameters, :idempotency_key, :context, 
:empirical_policy, :tags, :created_by, :infrastructure_pid, :auto_scheduled, :flow_id, :infrastructure_document_id, :parent_task_run_id, :state_id, :work_queue_id)]
```

Some examples:
- https://github.com/PrefectHQ/prefect/actions/runs/4414949522/jobs/7737303380 (has SQLITE debug logs)
- https://github.com/PrefectHQ/prefect/actions/runs/4369228293/jobs/7642782870

the test originally looks like this:
```
@pytest.mark.service("database")
@pytest.mark.flaky(max_runs=3)
async def test_orion_full_migration_works_with_data_in_db(sample_db_data):
    """
    Tests that downgrade migrations work when the database has data in it.
    """
    try:
        await run_sync_in_worker_thread(alembic_downgrade)
    finally:
        await run_sync_in_worker_thread(alembic_upgrade)
```

- In setup.cfg the pytest timeout is set to 60. 
- The primary error that happening here is that the test will often timeout because it takes longer than 60s to run 
- due to the nature of the test (actually running upgrades that modify the tables), it will timeout in the middle of the migrations leaving the database in an unexpected state
- then flaky will kick in and try to rerun the test including re-running the fixture and inserting data into the tables that no longer have their expected state (fully upgraded)

### Checklist

- [x] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`

### CODE REVIEW

**Comment by madkinsz** [comment]
Makes sense to me. It doesn't seem to make sense to have @flaky on a test that can run for 2 minutes. I'd love to get the runtime of this test down too.

**Review by madkinsz — APPROVED** [review]
